### DISCUSSION TITLE
Curious why Initialisms (ST1003) are disabled by default

### DISCUSSION THREAD

**Discussion by alexchao26** [dominikh/go-tools#953] [General]
This doesn't warrant opening an issue because it's [documented behavior](https://staticcheck.io/docs/options). Just curious as to why [ST1003](https://staticcheck.io/docs/checks#ST1003) is disabled by default. Is it just considered too opinionated?

I noticed this when updating the Go VSCode extension which now defaults to `staticcheck`, and these initialism warnings went away. Personally I like the initialism format as I got used to it from `golint`.

**Comment by dominikh** [comment]
Yeah, all of the checks in `ST*` that are disabled by default boil down to "this is too opinionated and/or too noisy for a lot of people". For ST1003, the issue is exacerbated by the fact that it doesn't just check for initialisms, but other identifier naming violations, too. For example, it complains about ALL_CAPS names for constants, which may be considered over the top by some people, and will conflict with some forms of code generation, especially C bindings.

**Reply by alexchao26** [reply]
Fair enough, I'm sure I'll get used to it in a month or two. Thanks!

**Reply by dominikh** [reply]
You should be able to configure the arguments passed to Staticcheck by VSCode and enable the check. You could also add a `staticcheck.conf` to your project, so that all people working on your project adhere to the same rules. See https://staticcheck.io/docs/#configuration
